src/sdi-notify.c

```diff
@@ -132,9 +146,9 @@ static void show_pending_update_notification(SdiNotify *self,
                                  NULL);
   notify_notification_add_action(notification, "default", _("Close"),
                                  app_close_notification, NULL, NULL);
-  IgnoreNotifyData *data = g_malloc(sizeof(IgnoreNotifyData));
-  data->self = self;
-  data->snaps = get_snap_list(snaps);
+  g_autoptr(GVariant) snap_list = get_snap_list(snaps);
+  // The memory block of data will be freed in the callback
```

Comment:
This is sometimes clearer by just doing it inline, i.e.
```
notify_notification_add_action(notification, "app.ignore-notification",
                                 _("Ignore"), app_ignore_snaps_notification,
                                 ignore_notify_data_new(self, snap_list), NULL);
```

Also, you can set the last argument to `ignore_notify_data_free` and then it doesn't have to be unreffed in the callback.

Reply:
good point. Done.